app/js/controllers/main.js

```diff
@@ -138,9 +181,13 @@ export default class MainController extends Controller {
       });
   }
 
-  _waitToBeClosed() {
-    Gesture.detect(this.closeGesture)
-      .then(() => this.view.close())
-      .then(() => this._waitToBeOpened());
+  close() {
+    if (!this._isOpen) {
```

Comment:
This could potentially get racy if we somehow trigger ```close()``` immediately after ```open()```, but I can't see that happening.